From da2a40d1970e83998de41a1b2694955290ef044e Mon Sep 17 00:00:00 2001
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Fri, 31 Dec 2021 15:59:39 +0800
Subject: [PATCH] kabi: reserve space for struct cpu_stop_work
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: da2a40d1970e83998de41a1b2694955290ef044e


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4JBL0
CVE: NA

-------------------------------

Reserve space for struct cpu_stop_work. Changing this struct will
affect set_cpus_allowed_ptr(), so reserve one kabi field.

Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/stop_machine.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/stop_machine.h b/include/linux/stop_machine.h
index 63ea9aff368f..08ec8e2fd9b2 100644
--- a/include/linux/stop_machine.h
+++ b/include/linux/stop_machine.h
@@ -6,6 +6,7 @@
 #include <linux/cpumask.h>
 #include <linux/smp.h>
 #include <linux/list.h>
+#include <linux/kabi.h>
 
 /*
  * stop_cpu[s]() is simplistic per-cpu maximum priority cpu
@@ -26,6 +27,7 @@ struct cpu_stop_work {
 	cpu_stop_fn_t		fn;
 	void			*arg;
 	struct cpu_stop_done	*done;
+	KABI_RESERVE(1)
 };
 
 int stop_one_cpu(unsigned int cpu, cpu_stop_fn_t fn, void *arg);
-- 
2.26.2

